modules/gateway/persist.go

```diff
@@ -78,3 +76,22 @@ func (g *Gateway) threadedSaveLoop() {
 		}()
 	}
 }
+
+// loadv033persist loads the v0.3.3 Gateway's persistent data from disk.
+func (g *Gateway) loadv033persist() error {
+	var nodes []modules.NetAddress
+	err := persist.LoadJSON(persist.Metadata{
+		Header:  "Sia Node List",
+		Version: "0.3.3",
+	}, &nodes, filepath.Join(g.persistDir, nodesFile))
+	if err != nil {
+		return err
+	}
+	for _, addr := range nodes {
+		err := g.addNode(addr)
+		if err != nil {
+			g.log.Printf("WARN: error loading node '%v' from persist: %v", addr, err)
+		}
+	}
+	return nil
+}
```

Comment:
Code looks good but it needs a test demonstrating that the gateway is able to load legacy persist correctly. My preference would be for using a file in the `testdata` dir and verifying it loads correctly. No reason not to just pull a gateway persist from an existing node.